### PR TITLE
Document current status of LVM in the Partitioner

### PR DESCRIPTION
**Pull request by ancorgs**
First step to implement correct support in YaST for all the LVM types/features that are currently detected by libstorage-ng.

See https://trello.com/c/P5XUCPLo/1879-3-follow-up-of-new-lvm-types-check-possible-impact-in-the-proposal-and-or-expert-partitioner-and-or-offline-upgrade

### CODE REVIEW

File: doc/lvm.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
+# LVM features and YaST
+
+The basic LVM functionality has been supported by YaST for ages. But LVM has a lot of features
+like RAID, snapshots, thin provisioning and whatsnot that often result in special types of logical
+volumes being present in the system.
+
+This document summarizes how well can YaST deal with systems that contain such advanced LVM setups.
```
Comment (aschnell) [inline doc/lvm.md:7]: 'can YaST' -> 'YaST can'

File: doc/lvm.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+operations it allows for each type of logical volume. At the current stage, some operations show an
+unexpected behavior and, in most cases, would need to be adjusted. That is represented in bold text.
+
+Note that, unlike RAID0, stripped LVs are not really a separate type. Many types of LVs can be
+stripped.
+
+### Normal LVM
+
+This is somehow the base case. Normal LVs are visualized in the standard way and all operations
+are supported.
```
Comment (aschnell) [inline doc/lvm.md:21]: Resizing it not allowed if the logical volume has snapshots and is active. Thus we decided to not allow it.

File: doc/lvm.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+- For thin pools
+  - Create: it works.
+  - Edit (format/mount): not allowed ("_Is a thin pool. It cannot be edited_").
+  - Resize: it works. Thin pools already in disk cannot be shrinked, which is correct.
+  - Delete: it works. Note it deletes the pool, all its thin volumes and the associated hidden LVs.
+
+- For thin LVs
+  - Create: it works. The **widgets for defining stripping are disabled and set to the default values**.
+    Maybe it would be better to show them disabled but with the pool values. Or to not show them at all.
+  - Edit (format/mount): not allowed ("_Is a thin pool. It cannot be edited_").
```
Comment (aschnell) [inline doc/lvm.md:46]: If that message is shows it is a serious bug.

File: doc/lvm.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+- Only the cache LV is displayed. No trace of the hidden cache volume.
+- **Nothing in the UI identifies the cache LV as being so**, just like the previous case.
+
+#### What can be done?
+
+- Create: not possible.
+- Edit (format/mount): just as a normal LV.
+- Resize: not allowed ("_Resizing of this type of LVM logical volumes is not supported_").
+- Delete: it works. It only deletes the cache LV, the LV used as cache survives (it becomes a
+  normal LV) **but is not visible immediately**. A reprobing after the commit phase is needed.
```
Comment (aschnell) [inline doc/lvm.md:80]: I consider the behaviour of LVM a bug and reported it, see https://bugzilla.suse.com/show_bug.cgi?id=1171907.

File: doc/lvm.md (hunk 3 above)
Comment (ancorgs) [reply doc/lvm.md:46]: Why? Not allowing to format/mount thin pools makes sense. Isn't it?

Comment (aschnell) [reply doc/lvm.md:46]: Yes, but it in the context of thin logical volumes.

Comment (ancorgs) [reply doc/lvm.md:46]: Oops, copy&paste error. I will fix it.

Comment (ancorgs) [reply doc/lvm.md]: Fixed.

File: doc/lvm.md (hunk 4 above)
Comment (ancorgs) [reply doc/lvm.md:83]: Link to bugzilla added.

File: doc/lvm.md (hunk 2 above)
Comment (ancorgs) [reply doc/lvm.md]: Added a note about that.

File: doc/lvm.md (hunk 1 above)
Comment (ancorgs) [reply doc/lvm.md]: I adapted the sentence.

File: doc/lvm.md [hunk 5]
```diff
@@ -18,7 +18,10 @@ stripped.
 ### Normal LVM
 
 This is somehow the base case. Normal LVs are visualized in the standard way and all operations
-are supported.
+are generally supported.
+
+The most noticeable exception is that YaST prevents resizing a logical volume that has snapshots and
+is active. It shows "_Resizing not supported since the logical volume has snapshots_".
```
Comment (aschnell) [inline doc/lvm.md:24]: YaST does not allow resizing even if the logical volume is inactive.

**Pull request merged**